### PR TITLE
Add PlaceholderAPI support

### PR DESCRIPTION
**Pull request by svnxd**

### CODE REVIEW

File: prison-spigot/src/main/java/tech/mcprison/prison/spigot/placeholder/PlaceHolderAPIIntegration.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	@Override
+	public String getIdentifier(){
+		return "Prison";
+	}
+
+	@Override
+	public String getVersion(){
+		return "1.0.0";
```
Comment (Perksey) [inline prison-spigot/src/main/java/tech/mcprison/prison/spigot/placeholder/PlaceHolderAPIIntegration.java]: Could we maybe hook this up to the one in plugin.yml via SpigotPrison? Is possible, but haven't done it in a while so it's worth googling.

File: prison-spigot/build.gradle [hunk 2]
```diff
@@ -35,14 +36,15 @@ dependencies {
...
-    compile 'org.inventivetalent.spiget-update:bukkit:1.4.0'
+    compile 'org.inventivetalent.spiget-update:bukkit:1.4.2-SNAPSHOT'
 
     compileOnly 'org.spigotmc:spigot-api:1.9.4-R0.1-SNAPSHOT'
     compileOnly 'net.milkbowl.vault:VaultAPI:1.6'
     compileOnly 'me.lucko.luckperms:luckperms-api:4.0'
     compileOnly('be.maximvdw:MVdWPlaceholderAPI:2.4.0-SNAPSHOT'){
         exclude group: 'org.spigotmc'
     }
+    compile 'me.clip:placeholderapi:2.9.1'
```
Comment (faizaand) [inline prison-spigot/build.gradle]:
```suggestion
    compileOnly 'me.clip:placeholderapi:2.9.1'
```

File: prison-spigot/src/main/java/tech/mcprison/prison/spigot/placeholder/PlaceHolderAPIIntegration.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	@Override
+	public String onRequest(OfflinePlayer player, String identifier){
+
+		if(identifier.equals("rank")){
+			return get(player.getUniqueId());
+		}
+
+		return null;
```
Comment (faizaand) [inline prison-spigot/src/main/java/tech/mcprison/prison/spigot/placeholder/PlaceHolderAPIIntegration.java]: Is returning null safe here?

Comment (svnxd) [reply prison-spigot/src/main/java/tech/mcprison/prison/spigot/placeholder/PlaceHolderAPIIntegration.java]: Yes, it just shows %prison_rank% wherever you use it.

**Comment by Perksey** [comment]
@faizaand follow-up review?

**Comment by Perksey** [comment]
Merging as all review comments have been rectified.

**Comment by rbluer** [comment]
This code was not tested on a clean severer, and as such it caused the prison plugin to fail.
It passed a simple compile with Gradle.  It passed appveyor. But caused a class not found exception to be thrown on plugin's onEnable.  I tested on 1.8.8 and 1.14.4 and confirmed the failure on both platforms.  Failures would have happened on other versions too.
I see both faizaand and perkesy asked if it was tested, bvnxd never confirmed that they tested it.
Would have been bad if someone tried to load the build artifact on their server.

**Pull request merged**